### DISCUSSION TITLE
How to support netcoreapp3.1 ?

### DISCUSSION THREAD

**Discussion by chrisbecke** [ThreeMammals/Ocelot#1883] [Q&A] [answered]
## Expected Behavior / New Feature

The documentation here, and on readthedocs, claims that netcoreapp3.1 and netcoreapp1 is supported. However, adding Ocelot to projects using these frameworks fails.

```bash
info : Installed Ocelot 17.0.0 from https://api.nuget.org/v3/index.json with content hash z6U2Vwn+M5ghzh2LdhTJApjhW/c1yeWTVlNPknWVome7X8Z+7B0DWHkX7l0+7ZVYnGPH0iHmgPPZJKCzF0xECg==.
error: NU1202: Package Ocelot 17.0.0 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Ocelot 17.0.0 supports: net5.0 (.NETCoreApp,Version=v5.0)
error: Package 'ocelot' is incompatible with 'all' frameworks in project '/Users/chrisb/Projects/demo/ocelot-test.csproj'.
```

## Steps to Reproduce the Problem

  1. `dotnet new -o ocelot-test -f netcoreapp3.1`
  1. `dotnet add ocelot-test package Ocelot`

## Specifications

  - Version: 17.0.0

**Comment by protang8** [comment, marked as answer]
I have the same issue.

I found the ocelot only support the ".NETCoreAPI,version=V5.0" or later from the "V16.0.2". 
So I install it with the "16.0.1", and it's success.

`Install-Package Ocelot -Version 16.0.1`

**Reply by raman-m** [reply]
Right! You can use old Ocelot versions for .NET Core 3.1
As a team, we don't support .NET Core 3.1 anymore!

**Comment by raman-m** [comment]
> @chrisbecke wrote [on Aug 2, 2021](https://github.com/ThreeMammals/Ocelot/discussions/1883#discussion-6034728)

Hi Chris!
Are you still with Ocelot? :wink: Sorry for the late reply!
We don't support .NET Core 3.1. I recommend you to start migration to .NET 6, 7 or even 8 frameworks.
